src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -227,15 +227,21 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
     TermControl::~TermControl()
     {
         _closing = true;
-        // Don't let anyone else do something to the buffer.
-        auto lock = _terminal->LockForWriting();
+        if (_initializedTerminal)
+        {
+            // Don't let anyone else do something to the buffer.
+            auto lock = _terminal->LockForWriting();

```

Comment:
What do these changes do?

Reply:
Wait, doesn't moving `lock` into the scope here make it unlock immediately on line 234?

Reply:
Yikes yea that's not great. I'll revert that - I was having some lifetime issues with TermControl where if you construct one, then don't add it to the UI, it never gets initialized, then it tries to destruct, and we explode here. 

But this change is definitely wrong.

Reply:
Actually the below line about the renderer not being null seems good to me, so I'm gonna keep it.